### PR TITLE
Added support for zend_oo data types

### PR DESCRIPTION
**Pull request by lander86**
Adding support for zend_oo data types on this branch.

### CODE REVIEW

File: lib/address.js
Comment (otoumas) [inline lib/address.js:9]: Missing ;

**Review by paolocappelletti — DISMISSED** [review]

File: test/transaction/transaction-sidechain.js
Comment (yurikoinaba) [inline test/transaction/transaction-sidechain.js:111]: Do you also want to check that the elements of the array are equal?

File: lib/transaction/transaction.js
Comment (yurikoinaba) [inline lib/transaction/transaction.js:48]: This is already defined 4 lines above.

File: lib/transaction/sidechain.js
Comment (MarcoOl94) [inline lib/transaction/sidechain.js:54]: add the new field **mcReturnAddress **

**Review by MarcoOl94 — CHANGES_REQUESTED** [review]

File: lib/transaction/transaction.js
Comment (yurikoinaba) [inline lib/transaction/transaction.js:426]: `this.version != -5` will always be `true` because we are in the `else` block

File: lib/networks.js
Comment (zain-cheng) [inline lib/networks.js:189]: Just curious, what is networkMagic?

File: test/block/block.js
Comment (otoumas) [inline test/block/block.js:55]: There is a lot of commented code in this file, can we get rid of it?

File: test/block/blockheader.js
Comment (otoumas) [inline test/block/blockheader.js:24]: unused var

File: test/transaction/sidechain-certificate.js
Comment (otoumas) [inline test/transaction/sidechain-certificate.js:1]: single quotes

File: test/transaction/sidechain-certificate.js
Comment (otoumas) [inline test/transaction/sidechain-certificate.js:13]: space after +

File: test/transaction/sidechain-certificate.js
Comment (otoumas) [inline test/transaction/sidechain-certificate.js:54]: use ===

File: test/transaction/sidechain-certificate.js
Comment (otoumas) [inline test/transaction/sidechain-certificate.js:55]: single quote

File: test/transaction/transaction-sidechain.js
Comment (otoumas) [inline test/transaction/transaction-sidechain.js:4]: single quotes

File: test/transaction/transaction-sidechain.js
Comment (otoumas) [inline test/transaction/transaction-sidechain.js:18]: ;

File: test/transaction/transaction-sidechain.js
Comment (otoumas) [inline test/transaction/transaction-sidechain.js:26]: space after +

**Review by otoumas — COMMENTED** [review]
A few comments 😅, most of them are just typos (missing ;, double/single quotes, === instead of ==)....
But in between all that noise there a re some open questions and posible changes to be made 😄

File: lib/block/blockheader.js
Comment (otoumas) [reply lib/block/blockheader.js:33]: nvm already solved this doubt with @MarcoOl94: 
version 4 keep reserved
version 3 change to scTxCommitment

File: lib/networks.js
Comment (MarcoOl94) [reply lib/networks.js:189]: NetworkMagic are bits that are used as a way to identify the separate messages sent between nodes on the network.

File: lib/script/script.js
Comment (otoumas) [reply lib/script/script.js:470]: chunk[6] should be an integer, we can either find an OP_code representing this or delete the line.

File: lib/script/script.js
Comment (otoumas) [reply lib/script/script.js:746]: Add to the datadoc of the method this difference with bitcoin: In zend the public keys are not sorted in multisig scripts

File: lib/script/script.js
Comment (otoumas) [reply lib/script/script.js:997]: Investigate it, the function is being called by transaction.js line 987 T`ransaction.prototype.addData = function(value) {`

File: lib/script/script.js
Comment (otoumas) [reply lib/script/script.js:1018]: This attributes are set in output.js and in input.js

File: lib/transaction/input/multisig.js
Comment (otoumas) [reply lib/transaction/input/multisig.js:28]: Add comment to the method doc as this is a difference with Bitcoin

File: lib/transaction/output.js
Comment (otoumas) [reply lib/transaction/output.js:58]: No, this is correct as it is a set function. We need to remove the return keyword

File: lib/transaction/sidechain.js
Comment (otoumas) [reply lib/transaction/sidechain.js:97]: unify to new buffer.Buffer

File: lib/transaction/sidechain.js
Comment (otoumas) [reply lib/transaction/sidechain.js:254]: Nope

File: lib/transaction/transaction.js
Comment (otoumas) [reply lib/transaction/transaction.js:121]: Used in block explorer

File: lib/transaction/transaction.js
Comment (otoumas) [reply lib/transaction/transaction.js:509]: Add a test and do the change

**Comment by zain-cheng** [comment]
From a high level, this looks fine to me.  I haven't gotten a chance to dig deeper.  @paolocappelletti Are you ready to approve the PR? Let me know if we need two approvers.

**Review by paolocappelletti — APPROVED** [review]

**Review by zain-cheng — APPROVED** [review]
@otoumas @MarcoOl94 I'm okay with merging these changes and then iterating on it.

**Pull request merged**
